src/main.cpp

```diff
@@ -1317,6 +1324,15 @@ MempoolReturnValue AcceptCertificateToMemoryPool(CTxMemPool& pool, CValidationSt
                 }
             }
 
+            // checking txs commitment tree validity
+            SidechainTxsCommitmentBuilder scCommitmentBuilder;
+            if (!scCommitmentBuilder.add(cert, view)) {
+                state.Invalid(
+                        error("%s(): SidechainTxsCommitmentBuilder failed\n", __func__),
+                                CValidationState::Code::INVALID, "bad-sc-cert-txscommitmenttree-fail");
+                return MempoolReturnValue::INVALID;
+            }
+
```

Comment:
Similar to what I mentioned in the previous comment, would it be possible to move this check to `Sidechain::checkCertSemanticValidity()`?

Reply:
See previous answer